codex/conf.nim

```diff
@@ -267,6 +274,21 @@ proc defaultDataDir*(): string =
 
   getHomeDir() / dataDir
 
+proc readValue*(r: var TomlReader, val: var EthAddress)
+               {.raises: [SerializationError, IOError, Defect].} =
+  val = EthAddress.init(r.readValue(string)).get()
+
+proc readValue*(r: var TomlReader, val: var SignedPeerRecord)
+               {.raises: [SerializationError, IOError, Defect].} =
+  let uri = r.readValue(string)
+  try:
+    if not val.fromURI(uri):
+      warn "Invalid SignedPeerRecord uri", uri=uri
+      quit QuitFailure
+  except CatchableError as exc:
+    warn "Invalid SignedPeerRecord uri", uri=uri, error=exc.msg
+    quit QuitFailure
```

Comment:
You could catch `SerializationError` and `IOError` using `without .. =? .., err` (catches all `CatchableErrors`) and  then and remove them from the `(up)raises` list. Also `parseCmdArg` has an override for `SignedPeerRecord`, no need to copy its contents:
```suggestion
proc readValue*(r: var TomlReader, val: var SignedPeerRecord)
               {.upraises: [].} =
  without uri =? r.readValue(string).catch, err:
    error "invalid SignedPeerRecord configuration value", error = err.msg
    quit QuitFailure
  
  val = SignedPeerRecord.parseCmdArg(uri)
```

Or if you'd prefer to catch `SerializationError` and `IOError` exceptions explicitly perhaps because they are raised in `readValue`:
```suggestion
proc readValue*(r: var TomlReader, val: var SignedPeerRecord)
               {.upraises: [].} =
  var uri = ""
  try:
    uri = r.readValue(string)
  except SerializationError, IOError:
    error "invalid SignedPeerRecord configuration value", error = getCurrentExceptionMsg()
  val = SignedPeerRecord.parseCmdArg(uri)
```